src/components/common/Pagination.component.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+
+import { PaginationNextButtonIconComponent } from '@/components/icon/PaginationNextButtonIcon.component.tsx';
+import { PaginationPreviousButtonIconComponent } from '@/components/icon/PaginationPreviousButtonIcon.component.tsx';
+
+import styles from './pagination.module.css';
+
+interface PaginationProps {
+    page: number;
+    limit: number;
+    total: number;
+    setCurrentPage: (page: number) => void;
+}
+
+export const PaginationComponent: React.FC<PaginationProps> = ({ page, limit, total, setCurrentPage }) => {
+    const totalPages = Math.ceil(total / limit);
+    const pages = Array.from({ length: totalPages }, (_, index) => index + 1);
+
+    const setPreviousPage = (pageNumber: number) => setCurrentPage(pageNumber <= 1 ? 1 : pageNumber - 1);
+    const setNextPage = (pageNumber: number) => setCurrentPage(pageNumber >= totalPages ? totalPages : pageNumber + 1);
+
+    return (
+        <nav>
+            <ul className={styles.paginationList}>
+                <li>
+                    <button
+                        className={`${styles.paginationButton} ${styles.paginationArrowButton}`}
+                        onClick={setPreviousPage.bind(null, page)}
```

Comment:
as an optional ehancement
```suggestion
                        onClick={() => setPreviousPage(page)}
```

Reply:
fixed